images/aws_upload.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash 
+
+set -e
+
+disk="$1"
+s3_bucket="cos-images"
+disk_name="cOS-Vanilla"
+disk_desc="cOS Vanilla Image"
+: "${github_sha:=none}"
+
+[ -f "${disk}" ] || exit 1
+
+echo "Uploading to S3 Bucket: ${s3_bucket}"
+aws s3 cp "${disk}" "s3://${s3_bucket}" --metadata "Name=${disk_name}"
+
+echo "Import snapshot to EC2"
+task_id=$(aws ec2 import-snapshot --description "${disk_desc}" \
+    --disk-container "Description=${disk_desc},Format=raw,UserBucket={S3Bucket=${s3_bucket},S3Key=$(basename ${disk})}" \
+    --tag-specifications "ResourceType=import-snapshot-task,Tags=[{Key=Name,Value=${disk_name}}]" \
+    | jq -r '.ImportTaskId')
+
+task_status=''
+
+# 10min timeout to import snapshot
+counter=0
+while [ ! "${task_status}" = "completed" ] && [ $counter -lt 60  ]; do
+    echo "Waiting for import-snapshot task to finalize. $((counter*10))s"
+    sleep 10
+    task_status=$(aws ec2 describe-import-snapshot-tasks \
+        --import-task-ids ${task_id} \
+        | jq -r '.ImportSnapshotTasks[0].SnapshotTaskDetail.Status')
+    counter=$((counter + 1))
+done
+
+[ ! "${task_status}" = "completed" ] && exit 1
+
+snap_id=$(aws ec2 describe-import-snapshot-tasks \
+    --import-task-ids ${task_id} \
+    | jq -r '.ImportSnapshotTasks[0].SnapshotTaskDetail.SnapshotId')
+
+echo "Tagging Snapshot"
+aws ec2 create-tags --resources "${snap_id}" \
+    --tags Key=Name,Value=${disk_name} Key=Project,Value=cOS Key=GITHUB_SHA,Value=$github_sha
```

Comment:
more interesting is the `Key=Flavor,Value=$FLAVOR` here, as it will allow us to create deletion policies based on the flavor images